modules/sdk-core/src/bitgo/utils/tss/eddsa/eddsa.ts

```diff
@@ -26,13 +28,64 @@ import {
 import { KeyShare, YShare } from './types';
 import baseTSSUtils from '../baseTSSUtils';
 import { KeychainsTriplet } from '../../../baseCoin';
-import assert from 'assert';
 
 /**
  * Utility functions for TSS work flows.
  */
 
 export class EddsaUtils extends baseTSSUtils<KeyShare> {
+  async verifyWalletSignatures(
+    userGpgPub: string,
+    backupGpgPub: string,
+    bitgoKeychain: Keychain,
+    decryptedShare: string,
+    verifierIndex: 1 | 2
+  ): Promise<void> {
+    assert(bitgoKeychain.commonKeychain);
+    assert(bitgoKeychain.walletHSMGPGPublicKeySigs);
+
+    const bitgoGpgKey = await getBitgoGpgPubKey(this.bitgo);
+
+    const userKeyPub = await openpgp.readKey({ armoredKey: userGpgPub });
+    const userKeyId = userKeyPub.keyPacket.getFingerprint();
+
+    const backupKeyPub = await openpgp.readKey({ armoredKey: backupGpgPub });
+    const backupKeyId = backupKeyPub.keyPacket.getFingerprint();
+
+    const walletSignatures = await openpgp.readKeys({ armoredKeys: bitgoKeychain.walletHSMGPGPublicKeySigs });
+    if (walletSignatures.length !== 2) {
+      throw new Error('Invalid wallet signatures');
+    }
+
+    if (userKeyId !== walletSignatures[0].keyPacket.getFingerprint()) {
```

Comment:
im not opposed to this check (it's a good sanity check), but what should *really* be checked is if `userKeyId` is equal to the `userKeyId=` in the notation data of the signatures from the HSM.    

https://github.com/BitGo/hsm-api/blob/44c9e8295407d6df411e547c63d38a07a9edc6e2/src/api/newsolanakey.js#L299-L314

That's the notation data.    This means that when the HSM certifies the user/backup gpg key, its signing packets that (in addition to standard pgp packets of certifying another key) _also_ include that data.

so just do what you did in `commonVerifyWalletSignature()` where you check against the raw notations.    `userKeyId !== Buffer.from(rawNotations[1].value).toString(),`

Reply:
This function is (transitively) calling `commonVerifyWalletSignature` and passing the user and backup key ids. This means that it's already checking what you are suggesting.
It first calls `verifyWalletSignature` for each signature and passes the user and backup key ids:
https://github.com/BitGo/BitGoJS/blob/0e6840e4b9a89aea30e784e0acede2377937fe6c/modules/sdk-core/src/bitgo/utils/tss/eddsa/eddsa.ts#L68-L86
`verifyWalletSignature` then calls `commonVerifyWalletSignature` and passes the same params in:
https://github.com/BitGo/BitGoJS/blob/0e6840e4b9a89aea30e784e0acede2377937fe6c/modules/sdk-core/src/bitgo/tss/eddsa/eddsa.ts#L413
The checks you are asking for are done here:
https://github.com/BitGo/BitGoJS/blob/0e6840e4b9a89aea30e784e0acede2377937fe6c/modules/sdk-core/src/bitgo/tss/common.ts#L111-L115

Reply:
Thanks the explanation.   resolving this.